store/tikv/region_cache_test.go

```diff
@@ -299,6 +299,28 @@ func (s *testRegionCacheSuite) TestRequestFail2(c *C) {
 	s.checkCache(c, 0)
 }
 
+func (s *testRegionCacheSuite) TestRegionEpochAheadOfTiKV(c *C) {
+	// Create a separated region cache to do this test.
+	pdCli := &codecPDClient{mocktikv.NewPDClient(s.cluster)}
+	cache := NewRegionCache(pdCli)
+
+	region := createSampleRegion([]byte("k1"), []byte("k2"))
+	region.meta.Id = 1
+	region.meta.RegionEpoch = &metapb.RegionEpoch{Version: 10, ConfVer: 10}
+	cache.insertRegionToCache(region)
+
+	r1 := metapb.Region{Id: 1, RegionEpoch: &metapb.RegionEpoch{Version: 9, ConfVer: 10}}
+	r2 := metapb.Region{Id: 1, RegionEpoch: &metapb.RegionEpoch{Version: 10, ConfVer: 9}}
+
+	bo := NewBackoffer(context.Background(), 2000000)
+
+	err := cache.OnRegionEpochNotMatch(bo, &RPCContext{Region: region.VerID()}, []*metapb.Region{&r1})
+	c.Assert(err, IsNil)
+	err = cache.OnRegionEpochNotMatch(bo, &RPCContext{Region: region.VerID()}, []*metapb.Region{&r2})
+	c.Assert(err, IsNil)
+	c.Assert(len(bo.errors), Equals, 2)
```

Comment:
Why the errors length is 2 ?

Reply:
One should occur at L317 and another should occur at L319.